### PR TITLE
Update gromacs to 2020.5 version.

### PR DESCRIPTION
**Pull request by longr**
Incorporating no MPI improvements and changes to the recipe from https://github.com/bioconda/bioconda-recipes/pull/25847 and bumping gromacs version to 2020.5
----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Review by stain — APPROVED** [review]

**Comment by stain** [comment]
@BiocondaBot please add label

**Comment by stain** [comment]
I think we forgot to increase the build number from 1, so may have to do a second PR

**Comment by stain** [comment]
@BiocondaBot please merge

File: recipes/gromacs/2020/meta.yaml [hunk 1]
```diff
@@ -23,7 +24,6 @@ requirements:
     - cmake
     - pocl
   host:
-    - openmp # [linux]
```
Comment (mabraham) [inline recipes/gromacs/2020/meta.yaml:26]: Why this change? OpenMP is arguably  more useful than MPI or OpenCL to bioconda users

**Comment by mabraham** [comment]
I've been attempting the rather unrelated #26703 and getting builds timing out after two hours. I didn't understand why until I saw this was newly merged. But I guess we may need to revert this until we get a build to pass

File: recipes/gromacs/2020/meta.yaml (hunk 1 above)
Comment (longr) [reply recipes/gromacs/2020/meta.yaml:26]: Thanks for spotting this, it is just a mistake. I was (to quickly) removing the changes brought in to enable MPI whilst trying to keep the non-mpi changes and deleted the wrong lines.

I thought the build time seemed odd, but when Stian re-ran the build it was fine.  I wonder what has caused the time out.  I shall try and look at this today, feel free to revert and I will do a new PR later.
